sigstore/verify/models.py

```diff
@@ -225,11 +227,30 @@ def from_bundle(
         certs = bundle.verification_material.x509_certificate_chain.certificates
         if len(certs) == 0:
             raise InvalidMaterials("expected non-empty certificate chain in bundle")
-        cert_pem = PEMCert(
-            load_der_x509_certificate(certs[0].raw_bytes)
-            .public_bytes(Encoding.PEM)
-            .decode()
-        )
+
+        # Per client policy in protobuf-specs: the first entry in the chain
+        # MUST be a leaf certificate, and the rest of the chain MUST NOT
+        # include a root CA or any intermediate CAs that appear in an
+        # independent root of trust.
+        #
+        # We expect some old bundles to violate the rules around root
+        # and intermediate CAs, so we issue warnings and not hard errors
+        # in those cases.
+        leaf_raw, *chain_raw = certs
+        leaf_cert = load_der_x509_certificate(leaf_raw.raw_bytes)
+        if not cert_is_leaf(leaf_cert):
+            raise InvalidMaterials(
+                "bundle contains an invalid leaf or non-leaf certificate in the leaf position"
+            )
+
+        for chain_cert_raw in chain_raw:
+            chain_cert = load_der_x509_certificate(chain_cert_raw.raw_bytes)
+            # TODO: We should also retrieve the root of trust here and
+            # cross-check against it.
+            if cert_is_root_ca(chain_cert):
+                logger.warning(
+                    "this bundle contains a root CA, making it subject to misuse"
+                )
```

Comment:
Flagging for review: we only produce a warning here, not an error. This is because, despite the `MUST NOT` in the bundle spec, some bundles might already exist with root and (redundant) intermediate CAs in them.

Reply:
Yep, good idea.

Reply:
Thinking about this a bit more: it'd be nice to have a stable URL to point to here. 

Do the `protobuf-specs` docstrings get rendered anywhere? CC @znewman01 @haydentherapper